Makefile.am

```diff
@@ -116,7 +116,8 @@ setup.h: $(srcdir)/setup.h.in Makefile
 	     -e 's|@prefix[@]|$(prefix)|g' \
 	     -e 's|@p4includedir[@]|$(p4includedir)|g' \
 	     -e 's|@p4_14includedir[@]|$(p4_14includedir)|g' \
-	$< >$@
+	$< >$@.tmp
+	@if cmp $@ $@.tmp; then rm $@.tmp; else mv $@.tmp $@; fi
```

Comment:
What situation is this protecting you from? It's a bit tricky to infer but my best guess is that you're running into a problem with building `$(BUILT_SOURCES)` at the same time as `all-am` in the new `check` rule. If so, I think the right solution here is just to not do that; see below.